packages/playground/assets/__tests__/assets.spec.ts

```diff
@@ -180,7 +181,7 @@ test('?raw import', async () => {
 })
 
 test('?url import', async () => {
-  const src = `console.log('hi')\n`
+  const src = readFile('foo.js')
   expect(await page.textContent('.url')).toMatch(
     isBuild
       ? `data:application/javascript;base64,${Buffer.from(src).toString(
```

Comment:
There is an actual problem here with importing Unicode characters when `isBuild` is true, which the test does not check for as it's only checking for the equivalence of the base64 encoded string.

When `foo.js` is changed to the following:
```js
console.log('hi 東京')
// newline character
```

It will produce `data:application/javascript;base64,Y29uc29sZS5sb2coJ2hpIOadseS6rCcpCg==`, which incorrectly outputs as the following due to assuming ISO-8859-1:
```js
console.log('hi æ±äº¬')
// newline character
```

We should instead specify the charset as `utf-8`
```
data:application/javascript;charset=utf-8;base64,Y29uc29sZS5sb2coJ2hpIOadseS6rCcpCg==
                            ^------------^
```

Which unfortunately requires a change not just in the test but in Vite as well, I believe.

However, the logic here for solving #2625 itself is sound, so it's not a blocker for this PR.

Reply:
Good catch, let's create a new issue for that one 👍🏼